### PR TITLE
Update Go version to 1.21 from 1.19

### LINKED ISSUE #36: Upgrade to go version 1.21
Upgrade the config file validator to go version 1.21. As part of the update the following code revisions should be made:

* Replace the [isExtensionIncluded](https://github.com/Boeing/config-file-validator/pull/35/files#diff-a65676dbf68cd041d9bfecd616a578a7fbd7a2f8a85656b073281acca0026af4R120) function with `slice.contains`

Also the following updates need to be made to the repo:

* Update the Dockerfile to use go:1.21 (also go ahead and update Apline Linux version too)
* Update the go.mod to specifu go:1.21
* Update the version of go used by the Github Action runner
* Update the release Github Action runner

### PR DESCRIPTION
**Pull request by sandyydk**
**Changes in this PR**

- Update Go version to 1.21 from 1.19.  
- Replace IsExtensionExcluded with slices.Contains
- Update go versions in workflow and Dockerfile

resolves #36

### CODE REVIEW

File: pkg/finder/fsfinder.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ import (
 	"strings"
 
 	"github.com/Boeing/config-file-validator/pkg/filetype"
+	"golang.org/x/exp/slices"
```
Comment (kehoecj) [inline pkg/finder/fsfinder.go:10]: You should be able to import slices like a normal standard library import

```suggestion
        "slices"
	"strings"

	"github.com/Boeing/config-file-validator/pkg/filetype"
```

File: pkg/finder/fsfinder.go [hunk 2]
```diff
@@ -116,13 +117,6 @@ func (fsf FileSystemFinder) Find() ([]FileMetadata, error) {
 }
 
 // isExtensionExcluded returns true if extension exists in exclude list.
-// TODO: refactor to slices.Contains when project will be updated to go 1.21.
 func (fsf FileSystemFinder) isExtensionExcluded(ext string) bool {
```
Comment (kehoecj) [inline pkg/finder/fsfinder.go:120]: This entire function can be deleted now with the slices library supporting contains. The call to this function on line 94 can be replaced with `slices.Contains[[]string](fsf.ExcludeFileTypes, ext)`

Make sure to remove any unit tests that were written for `isExtensionExcluded`

**Review by kehoecj — CHANGES_REQUESTED** [review]
Looks good - a couple comments to address

File: pkg/finder/fsfinder.go (hunk 2 above)
Comment (sandyydk) [reply pkg/finder/fsfinder.go]: @kehoecj I did exactly this in the first commit. Reverted thinking we may need to keep the abstraction alive. Updated again. Thanks.

File: pkg/finder/fsfinder.go (hunk 1 above)
Comment (sandyydk) [reply pkg/finder/fsfinder.go]: Sure. My linter was not allowing this. Sure. Can you trigger the GitHub Action?

**Review by kehoecj — APPROVED** [review]
LGTM

**Review by jackswiney — APPROVED** [review]

**Comment by kehoecj** [comment]
Merged! Thank you @sandyydk for the contribution 💯

**Pull request merged**